File: backend/src/main/java/ca/metricalsky/winston/entity/view/AuthorChannelView.java
package ca.metricalsky.winston.entity.view;

import java.time.OffsetDateTime;

public interface AuthorChannelView {

    String getChannelTitle();

    String getChannelHandle();

    Long getVideoCount();

    Long getTotalCommentCount();

    Long getReplyCount();

    OffsetDateTime getFirstCommentedAt();

    OffsetDateTime getLastCommentedAt();

    default Long getCommentCount() {
        return getTotalCommentCount() - getReplyCount();
    }
}

File: database/src/main/kotlin/ca/metricalsky/winston/database/view/AuthorChannelView.kt
package ca.metricalsky.winston.database.view

import java.time.OffsetDateTime

interface AuthorChannelView {

    fun getChannelTitle(): String

    fun getChannelHandle(): String

    fun getVideoCount(): Long

    fun getTotalCommentCount(): Long

    fun getReplyCount(): Long

    fun getFirstCommentedAt(): OffsetDateTime

    fun getLastCommentedAt(): OffsetDateTime

    fun getCommentCount(): Long {
        return getTotalCommentCount() - getReplyCount()
    }
}
